core/src/main/java/io/fabric8/maven/core/util/SpringBootUtil.java

```diff
@@ -80,25 +93,79 @@ protected static Properties getPropertiesResource(URL resource) {
     /**
      * Returns a {@code Properties} representation of the given Yaml file on the project classpath if found or an empty properties object if not
      */
-    public static Properties getPropertiesFromYamlFile(MavenProject project, String yamlFileName) {
+    public static Properties getPropertiesFromYamlFile(MavenProject project, String yamlFileName,
+                                                       List<String> activeProfiles) {
         URLClassLoader compileClassLoader = MavenUtil.getCompileClassLoader(project);
         URL resource = compileClassLoader.findResource(yamlFileName);
-        return getPropertiesFromYamlResource(resource);
+        return getPropertiesFromYamlResource(resource, activeProfiles);
     }
 
     /**
      * Returns a {@code Properties} representation of the given Yaml resource or an empty properties object if the resource is null
      */
-    protected static Properties getPropertiesFromYamlResource(URL resource) {
+    protected static Properties getPropertiesFromYamlResource(URL resource, List<String> activeProfiles) {
         if (resource != null) {
             try (InputStream yamlStream = resource.openStream()) {
-                Yaml yaml = new Yaml();
-                @SuppressWarnings("unchecked")
-                SortedMap<String, Object> source = yaml.loadAs(yamlStream, SortedMap.class);
+                Yaml yaml = new Yaml(new SafeConstructor());
+
+                Map<String, Map> profileDocs = new HashMap<>();
+
                 Properties properties = new Properties();
-                if (source != null) {
-                    properties.putAll(getFlattenedMap(source));
+
+                Iterable<Object> yamlDoc = yaml.loadAll(yamlStream);
+
+                Iterator yamlDocIterator = yamlDoc.iterator();
+
+                int docCount = 0;
+                while (yamlDocIterator.hasNext()) {
+                    Map docRoot = (Map) yamlDocIterator.next();
+
+                    String profiles = null;
+
+                    if (docRoot.containsKey("spring")) {
+
+                        LinkedHashMap value = (LinkedHashMap) docRoot.get("spring");
+
+                        Object profilesValue = value.get("profiles");
+
+                        if (profilesValue instanceof Map) {
+                            Map profileMap = (Map) profilesValue;
+                            if (activeProfiles.isEmpty() && docCount > 0) {
+                                if (profileMap.containsKey("active")) {
+                                    activeProfiles.addAll(getActiveProfiles((String) profileMap.get("active")));
+                                }
+                            }
+                        } else if (profilesValue instanceof String) {
+                            profiles = (String) profilesValue;
+                        }
+                    }
+
+                    if (profiles != null) {
+                        String[] profileSplit = profiles.split("\\s*,\\s*");
```

Comment:
If profiles can take multiple values, why do you split it here then ? And couldn't it be that `profiles` could be also a list ?